src/routes/Wallets/modules/wallets.js

```diff
@@ -1,234 +1,133 @@
-  type: string,
-  payload: {
-    activeWalletId: ?WalletId,
-  },
-} => ({
-  type: SET_ACTIVE_WALLET_ID,
+export const setActiveWallet = (activeWalletId: ?WalletId): WalletsSetActiveWalletAction => ({
+  type: SET_ACTIVE_WALLET,
   payload: {
     activeWalletId,
   },
 })
 
-export const setInvalidField = (fieldName: string, message: string): {
-  type: string,
-  payload: {
-    fieldName: string,
-    message: string,
-  },
-} => ({
-  type: SET_INVALID_FIELD,
-  payload: {
-    fieldName,
-    message,
-  },
-})
-
-export const clean = (): { type: string } => ({
+export const clean = (): WalletsCleanAction => ({
   type: CLEAN,
 })
 
-const initialState: WalletsData = {
-  invalidFields: {},
+const initialState: WalletsState = {
   items: [],
-  password: '',
-  toggledWalletId: null,
-  showActionsWalletId: null,
+  passwordHint: '',
   activeWalletId: null,
-  walletAction: null,
-  isInitialised: false,
+  toggledWalletId: null,
+  passwordOptions: null,
+  mnemonicOptions: null,
+  testPasswordData: null,
 }
 
 const wallets = (
-  state: WalletsData = initialState,
-  action: FSA,
-): Object => {
-  const { type, payload }: FSA = action
-
-  switch (type) {
-    case INIT_FINISH: {
-      return assoc('isInitialised', true)(state)
-    }
-
-    case SET_WALLETS: {
-      return assoc('items', payload.items)(state)
-    }
-
-    case TOGGLE_WALLET: {
-      return compose(
-        assoc('toggledWalletId', payload.toggledWalletId),
-        assoc('showActionsWalletId', null),
-      )(state)
-    }
-
-    case SHOW_ACTIONS_MENU: {
-      return compose(
-        assoc('toggledWalletId', null),
-        assoc('showActionsWalletId', payload.showActionsWalletId),
-      )(state)
-    }
-
-    case SET_WALLET_ACTION: {
-      return assoc('walletAction', payload.walletAction)(state)
-    }
-
-    case SET_PASSWORD: {
-      return compose(
-        assoc('password', payload.password),
-        assocPath(['validFields', 'password'], null),
-        assocPath(['invalidFields', 'password'], null),
-      )(state)
-    }
-
-    case SET_ACTIVE_WALLET_ID: {
-      return assoc('activeWalletId', payload.activeWalletId)(state)
-    }
-
-    case SET_INVALID_FIELD: {
-      return assocPath(['invalidFields', payload.fieldName], payload.message)(state)
-    }
-
-    case CLEAN: {
-      return compose(
-        assoc('password', ''),
-        assoc('walletAction', null),
-        assoc('toggledWalletId', null),
-        assocPath(['validFields', 'password'], null),
-        assocPath(['invalidFields', 'password'], null),
-      )(state)
-    }
-
-    default: return state
+  state: WalletsState = initialState,
+  action: WalletsAction,
+): WalletsState => {
+  switch (action.type) {
+    case SET_WALLETS:
+      return {
+        ...state,
+        ...action.payload,
+      }
+
+    case TOGGLE_WALLET:
+      return {
+        ...state,
+        toggledWalletId: action.payload.toggledWalletId,
```

Comment:
I think here there will be better to use
```
const { toggledWalletId } = action.payload

return {
...state,
toggledWalletId 
}
```

Reply:
@flexxnn in your case, we should change `case` block:
from
```
case TYPE:
  return ...
```
to
```
case TYPE: {
 ...
}
```
if all `case`'s without curly brackets - better don't use destructuring for only one param